src/billing-form/billing-form.component.tsx

```diff
@@ -295,63 +153,53 @@ const BillingForm: React.FC<BillingFormProps> = ({ patientUuid, closeWorkspace }
-                    <TableCell id={row.Item + 'Total'} className="totalValue">
-                      {row.Total}
-                    </TableCell>
-                    <TableCell>
-                      <TrashCan onClick={() => removeItemFromBill(row.uuid)} className={styles.removeButton} />
-                    </TableCell>
-                  </TableRow>
-                ))
-              ) : (
-                <p>{t('loading', 'Loading...')}</p>
-              )}
+              {selectedItems.map((row) => (
+                <TableRow>
+                  <TableCell>{row.display}</TableCell>
+                  <TableCell>
+                    <NumberInput
+                      id={row.uuid}
+                      min={1}
+                      value={row.quantity}
+                      onChange={(_, { value }) => {
+                        const number = parseFloat(String(value));
+                        updateQuantity(row.uuid, isNaN(number) ? 1 : number);
+                      }}
+                    />
+                  </TableCell>
+                  <TableCell id={row.uuid + 'Price'}>{row.price}</TableCell>
+                  <TableCell id={row.uuid + 'Total'} className="totalValue">
+                    {row.price * row.quantity}
+                  </TableCell>
+                  <TableCell>
+                    <TrashCan className={styles.removeButton} onClick={() => removeSelectedBillableItem(row.uuid)} />
+                  </TableCell>
+                </TableRow>
+              ))}
               <TableRow>
                 <TableCell colSpan={3}></TableCell>
                 <TableCell style={{ fontWeight: 'bold' }}>{t('grandTotal', 'Grand total')}:</TableCell>
-                <TableCell id="GrandTotalSum">{convertToCurrency(grandTotal, defaultCurrency)}</TableCell>
+                <TableCell id="GrandTotalSum">
+                  {convertToCurrency(calculateTotalAmount(selectedItems), defaultCurrency)}
+                </TableCell>
               </TableRow>
             </TableBody>
           </Table>
-        </Stack>
-
-        <ButtonSet className={isTablet ? styles.tablet : styles.desktop}>
-          <Button className={styles.button} kind="secondary" disabled={isSubmitting} onClick={closeWorkspace}>
-            {t('discard', 'Discard')}
-          </Button>
-          <Button
-            className={styles.button}
-            kind="primary"
-            onClick={postBillItems}
-            disabled={isSubmitting || saveDisabled}
-            type="submit">
-            {isSubmitting ? (
-              <InlineLoading description={t('saving', 'Saving') + '...'} />
-            ) : (
-              t('saveAndClose', 'Save and close')
-            )}
-          </Button>
-        </ButtonSet>
+        )}
       </div>
+
+      <ButtonSet className={isTablet ? styles.tablet : styles.desktop}>
+        <Button className={styles.button} kind="secondary" disabled={isSubmitting} onClick={closeWorkspace}>
```

Comment:
```suggestion
        <Button className={styles.button} kind="secondary" disabled={isSubmitting || selectedItems.length === 0} onClick={closeWorkspace}>
```

Minor thing, but if we want to prevent empty submissions, maybe we could do something like this (in lieu of validation logic backed by a zod schema).